### PR TITLE
Fix ui slice issue

### PR DESCRIPTION
**Pull request by zilongshanren**
fix issue: https://github.com/cocos2d/cocos2d-x/issues/17654

### CODE REVIEW

File: cocos/2d/CCSprite.cpp [hunk 1]
```diff
@@ -1654,7 +1654,6 @@ void Sprite::setBatchNode(SpriteBatchNode *spriteBatchNode)
 
     // self render
     if( ! _batchNode ) {
-        _renderMode = RenderMode::QUAD;
```
Comment (minggo) [inline cocos/2d/CCSprite.cpp:1657]: why modify the render mode?

**Comment by tiagomartines11** [comment]
Thank you for looking into this.

After apply this PR, I am getting a crash on `TMXTiledMap::create` 

image

This is the line at fault, probably because `_renderMode` has invalid data, so it tries to access a null `_textureAtlas`:

`Texture2D *tex = (_renderMode == RenderMode::QUAD_BATCHNODE) ? _textureAtlas->getTexture() : _texture;`

**Comment by zilongshanren** [comment]
@tiagomartines11 
thanks, I will fix it.

**Comment by tiagomartines11** [comment]
It seems to be working properly now :) Thanks!

**Pull request merged**